web/app/Http/Controllers/PageController.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Wikijump\Http\Controllers;
+
+use Illuminate\Contracts\View\View;
+use Illuminate\Support\Facades\URL;
+use Wikidot\Utils\GlobalProperties;
+use Wikijump\Helpers\LegacyTools;
+
+class PageController extends Controller
+{
+    /** Returns a `view` for the current page. */
+    public function show(): View
+    {
+      $values = LegacyTools::generateScreenVars();
+
+      $title = null;
+      $canonical = URL::current();
+      $license = null;
+      $social_title = null;
+      $sidebar_content = null;
+      $page_content = null;
+      $category = null;
+      $breadcrumbs = null;
+      $page_title = null;
+      $revision = null;
+      $timestamp = null;
+      $tags = null;
+
+      if ($values['site']) {
+        $title = $values['site']->getName();
+      }
+
+      if ($values['wikiPage']) {
+          $page = $values['wikiPage'];
+
+          $sidebar_content = $values['sideBar1Content'] ?? null;
+          $page_content = $values['pageContent'] ?? null;
+          $tags = $values['tags'] ?? null;
+          $revision = $page->getRevisionNumber();
+          $timestamp = $page->getDateLastEdited()->getTimestamp();
+          $page_title = $page->getTitleOrUnixName();
+          $title = $page_title . ' | ' . $title;
```

Comment:
Since this is only composed of static strings and variables, you can use interpolation:
```suggestion
          $title = "$page_title | $title";
```

Reply:
...is that why every variable in PHP starts with `$`? lmao

I'll change the code here and in other places where its relevant